rudder-core/src/main/scala/com/normation/rudder/services/nodes/NodeInfoService.scala

```diff
@@ -120,18 +153,62 @@ trait NodeInfoService {
    * @return
    */
   def getAllSystemNodeIds() : Box[Seq[NodeId]]
+
+
+  /**
+   * Getting something like a nodeinfo for pending / deleted nodes
+   */
+  def getPendingNodeInfos(): Box[Map[NodeId, NodeInfo]]
```

Comment:
Would it be easier if we had "getAll( state : InventoryStatus)" and refactor some logic that is the same for the three states ?